articles/rules-are-dangerous.md

```diff
@@ -2,45 +2,45 @@
-* The dependency inversion principle says “Abstractions should not depend on details. Details should depend on abstractions". I have seen projects that tried to follow this principle very strictly and everywhere in their software. They therefore totally avoided to use any concrete class directly as much as possible. As a result, there were a lot of interfaces with only one implementation class each and many factories to create instances of those classes. This resulted in a code base that is much more complex than needed. It makes the software difficult to understand and maintain without bringing any benefit of really needed flexibility into that software.
-* People read the book about “Clean Code” that says that many code comments can be avoided by writing self-explaining code instead. While this is true for many cases the rule is often miss-used by developers for arguing not to write any comments at all in the source code. But there are many things that you cannot explain by using good method or variable names only. E.g. why you had to implement something like that. For this purpose, good comments are still very valuable and should not be denied. I would say that a well put comment can explain much more than hundred good variable names.
-* The article by Martin Fowler about the “Page Object Pattern” once said to never put assertions into page objects. Someone read that article and wanted to convince the whole project team that it is very important to follow this rule very strictly. The arguments for the rule in the blog post were not very convincing to me. I brought up many examples where breaking that rule caused much more benefit than sticking to it. Nevertheless I was not taken serious. Since Martin Fowler wrote it, it counts as if it was written in the bible. Nobody wanted to constructively argue about it anymore. Fun fact: In the meantime Martin Fowler changed his article to mention both opinions on this topic. This at least let’s the option open to the reader to decide what fits best in his concrete context. Unfortunately, not all famous technical opinion makers are so wise as Mr. Fowler.
+* The dependency inversion principle says “Abstractions should not depend on details. Details should depend on abstractions". I have seen projects that tried to follow this principle very strictly and everywhere in their software. They therefore totally avoided directly using any concrete class as much as possible. As a result, there were a lot of interfaces with only one implementation class each and many factories to create instances of those classes. This resulted in a code base that is much more complex than needed. It makes the software difficult to understand and maintain without bringing any benefit of much- needed flexibility into that software. 
```

Comment:
What is "much- needed" - this looks weird, what I meant was "no benefit of flexibility that is at all needed" (Deutsch: keine Flexibilität die tatsächlich gebraucht wird)

Reply:
ok

Reply:
https://github.com/Zuehlke/fifty-shades/commit/36b464815149809af537129d854954e1e795eb1d